automated_analysis.py

```diff
@@ -222,197 +211,23 @@
-                assert cc.coding_mode == CodingModes.MULTIPLE, "Other CodingModes not (yet) supported"
-                for label in td[cc.coded_field]:
-                    code = cc.code_scheme.get_code_with_code_id(label["CodeID"])
-                    if code.control_code == Codes.STOP:
-                        continue
-                    themes[f"{cc.analysis_file_key}_{code.string_value}"]["Total Participants"] += 1
-                    update_survey_counts(themes[f"{cc.analysis_file_key}_{code.string_value}"], td)
-                    if code.code_type == CodeTypes.NORMAL:
-                        relevant_participant = True
-
-            if relevant_participant:
-                themes["Total Relevant Participants"]["Total Participants"] += 1
-                update_survey_counts(themes["Total Relevant Participants"], td)
-
-        set_survey_percentages(themes["Total Relevant Participants"], themes["Total Relevant Participants"])
-
-        for cc in episode_plan.coding_configurations:
-            assert cc.coding_mode == CodingModes.MULTIPLE, "Other CodingModes not (yet) supported"
-
-            for code in cc.code_scheme.codes:
-                if code.code_type != CodeTypes.NORMAL:
-                    continue
-
-                theme = themes[f"{cc.analysis_file_key}_{code.string_value}"]
-                set_survey_percentages(theme, themes["Total Relevant Participants"])
-
+        theme_distributions.export_theme_distributions_csv(
+            individuals, CONSENT_WITHDRAWN_KEY,
+            coding_plans_to_analysis_configurations(PipelineConfiguration.DEMOG_CODING_PLANS),
+            [],
+            f
+        )
+
+    log.info("Computing theme distributions...")
     with open(f"{automated_analysis_output_dir}/theme_distributions.csv", "w") as f:
-        headers = ["Question", "Variable"] + list(make_survey_counts_dict().keys())
-        writer = csv.DictWriter(f, fieldnames=headers, lineterminator="\n")
-        writer.writeheader()
-
-        last_row_episode = None
-        for episode, themes in episodes.items():
-            for theme, survey_counts in themes.items():
-                row = {
-                    "Question": episode if episode != last_row_episode else "",
-                    "Variable": theme,
-                }
-                row.update(survey_counts)
-                writer.writerow(row)
-                last_row_episode = episode
+        theme_distributions.export_theme_distributions_csv(
+            individuals, CONSENT_WITHDRAWN_KEY,
+            coding_plans_to_analysis_configurations(PipelineConfiguration.RQA_CODING_PLANS),
+            coding_plans_to_analysis_configurations(PipelineConfiguration.SURVEY_CODING_PLANS),
+            f
+        )
 
     # Export raw messages labelled with Meta impact, gratitude and about conversation programmatically known as impact/success story
     log.info("Exporting success story raw messages for each episode...")
```

Comment:
It should be possible to use sample_messages in Core to do this bit. If not, I hope we can update the implementation of sample_messages so we can.

Reply:
Thanks forgot this here.

Reply:
updated to use sample_messagea in Core.